front/form.form.php

```diff
@@ -142,21 +142,13 @@
 
       // redirect to created item
       if ($_SESSION['glpibackcreated']) {
-         $item_ticket = new Item_Ticket;
-         $tickets = $item_ticket->find([
-            'itemtype' => PluginFormcreatorFormAnswer::class,
-            'items_id' => $formAnswer->getID(),
-         ]);
-
-         if (count($tickets) === 1) {
-            $current_ticket = array_pop($tickets);
-            $tickets_id = $current_ticket['tickets_id'];
-            Html::redirect(Ticket::getFormURLWithID($tickets_id));
-         }
-
-         if (count($tickets) > 1) {
-            Html::redirect(PluginFormcreatorFormAnswer::getFormURLWithID($formAnswer->getID()));
+         print_r($formAnswer->targetList);
+         if (count($formAnswer->targetList) == 1) {
+            $target = current($formAnswer->targetList);
+            $url = call_user_func_array([$target['itemtype'], "getFormURLWithID"], [$target['items_id']]);
```

Comment:
This suggestion is related to the other suggestion below.

```suggestion
            $url = $target->getFormURLWithID($target->getID());
```